modules/videoio/include/opencv2/videoio.hpp

```diff
@@ -719,6 +719,25 @@ enum VideoCaptureOBSensorProperties{
 
 //! @} videoio_flags_others
 
+/** @brief Read data stream interface
+ */
+class CV_EXPORTS_W IReadStream
```

Comment:
@opencv-alalek, @asmorkalov , need your opinion on naming as we ready to move it to the public API. There are `VideoCapture`, [IStreamSource ](https://docs.opencv.org/4.x/da/d03/classcv_1_1gapi_1_1wip_1_1IStreamSource.html), [VideoReader](https://docs.opencv.org/4.x/db/ded/classcv_1_1cudacodec_1_1VideoReader.html), [VideoWriter](https://docs.opencv.org/4.x/dd/d9e/classcv_1_1VideoWriter.html). So does it make sense to keep the order and name it `IStreamReader`? Moreover, it matches PyTorch's naming: https://docs.opencv.org/4.x/dd/d9e/classcv_1_1VideoWriter.html